File: web_viewer/backend/schemas/project.py
"""Pydantic schemas for Project endpoints."""

from datetime import datetime

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    """Request to create a new project/case."""

    name: str
    description: str | None = None
    patient_id: str | None = None
    reconstruction_type: str | None = None
    implant_material: str | None = None
    notes: str | None = None
    region_code: str | None = None
    metadata_json: str | None = None


class ProjectUpdate(BaseModel):
    """Request to update a project/case."""

    name: str | None = None
    description: str | None = None
    patient_id: str | None = None
    reconstruction_type: str | None = None
    implant_material: str | None = None
    notes: str | None = None
    region_code: str | None = None
    metadata_json: str | None = None


class ProjectRead(BaseModel):
    """Response schema for project/case."""

    id: str
    name: str
    description: str | None = None
    patient_id: str | None = None
    reconstruction_type: str | None = None
    implant_material: str | None = None
    notes: str | None = None
    region_code: str | None = None
    metadata_json: str | None = None
    created_at: datetime
    updated_at: datetime
    created_by: str

    model_config = {"from_attributes": True}

File: web_viewer/frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description: string | null;
  patient_id: string | null;
  reconstruction_type: string | null;
  implant_material: string | null;
  notes: string | null;
  region_code: string | null;
  metadata_json: string | null;
  created_at: string;
  updated_at: string;
  created_by: string;
}

export interface ProjectCreate {
  name: string;
  description?: string;
  patient_id?: string;
  reconstruction_type?: string;
  implant_material?: string;
  notes?: string;
  region_code?: string;
  metadata_json?: string;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  patient_id?: string;
  reconstruction_type?: string;
  implant_material?: string;
  notes?: string;
  region_code?: string;
  metadata_json?: string;
}
